### PR TITLE
feat: share comment feature

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Introducing a way to determine comment position (and scroll to it)
- Share comment start (awaiting design)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| New | share post | extra: { provider: ShareProvider, origin: Origin, commentId } |
| New | open post | extra: { origin: Origin, commentId } |
| New | close post | extra: { origin: Origin, commentId } |

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [x] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-250 #done

### CODE REVIEW

**Review by kirillkurko — APPROVED** [review]
Looks good to me, left only one general question

File: packages/extension/src/companion/CompanionDiscussion.tsx [hunk 1]
```diff
@@ -34,6 +42,7 @@ export function CompanionDiscussion({
     onInput,
     parentComment,
   } = useCompanionPostComment(post);
+  const [shareComment, setShareComment] = useState<Comment>();
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/extension/src/companion/CompanionDiscussion.tsx:45]: Good point, we can set it to null as the default would that work?
We can also check to make this a custom hook, but it seemed like overkill for now.

[the remark this answers is not in the document]
Comment (kirillkurko) [reply packages/extension/src/companion/CompanionDiscussion.tsx:45]: Yeah, custom hook is too much, setting initial value to `null` is a good solution. And I think in the future we may want to change linter to throw at least warning for such cases

**Review by sshanzel — APPROVED** [review]
Nice work on this! We finally have the scroll to comment feature 🚢

File: packages/extension/src/companion/CompanionDiscussion.tsx [hunk 2]
```diff
@@ -57,6 +66,7 @@ export function CompanionDiscussion({
           post={post}
           applyBottomMargin={false}
           onClick={onCommentClick}
+          onShare={(comment) => setShareComment(comment)}
```
Comment (idoshamun) [inline packages/extension/src/companion/CompanionDiscussion.tsx:69]:
```suggestion
          onShare={setShareComment}
```

File: packages/shared/src/components/widgets/SocialShare.tsx [hunk 3]
```diff
@@ -21,78 +21,82 @@ import TelegramIcon from '../icons/Telegram';
...
   post,
+  comment,
   origin,
   columns,
   column,
   row,
 }: SocialShareProps & FeedItemPosition): ReactElement => {
+  const isComment = !!comment;
+  const link = isComment
+    ? `${post?.commentsPermalink}#c-${comment.id}`
```
Comment (idoshamun) [inline packages/shared/src/components/widgets/SocialShare.tsx:41]: I think you should create a function that hashes the comment. I see you use it many times `c-${id}`. It's better to store it somewhere

**Review by idoshamun — APPROVED** [review]
A few minor coding suggestions, analytics looks great

**Comment by rebelchris** [comment]
@idoshamun Great suggestion for the new comment popup, i'll include that in this PR.

**Review by idoshamun — APPROVED** [review]
Looks great!

**Pull request merged**